waku/v2/utils/peers.nim

```diff
@@ -16,15 +17,73 @@ import
           multicodec,
           peerid,
           peerinfo,
+          peerstore,
           routing_record]
 
+#import
+#  ../node/peer_manager/waku_peer_store
+# todo organize this
+
+type
+  Connectedness* = enum
+    # NotConnected: default state for a new peer. No connection and no further information on connectedness.
+    NotConnected,
+    # CannotConnect: attempted to connect to peer, but failed.
+    CannotConnect,
+    # CanConnect: was recently connected to peer and disconnected gracefully.
+    CanConnect,
+    # Connected: actively connected to peer.
+    Connected
+
+  PeerOrigin* = enum
+    UnknownOrigin,
+    Discv5,
+    Static,
+    Dns
+
+  PeerDirection* = enum
+    UnknownDirection,
+    Inbound,
+    Outbound
+
+  # Keeps track of the Connectedness state of a peer
+  ConnectionBook* = ref object of PeerBook[Connectedness]
+
+  # Last failed connection attemp timestamp
+  LastFailedConnBook* = ref object of PeerBook[Moment]
+
+  # Failed connection attempts
+  NumberFailedConnBook* = ref object of PeerBook[int]
+
+  # Keeps track of when peers were disconnected in Unix timestamps
+  DisconnectBook* = ref object of PeerBook[int64]
+
+  # Keeps track of the origin of a peer
+  SourceBook* = ref object of PeerBook[PeerOrigin]
+
+  # Direction
+  DirectionBook* = ref object of PeerBook[PeerDirection]
+
+  # ENR Book
+  ENRBook* = ref object of PeerBook[enr.Record]
```

Comment:
Even though the `peers` util module itself should probably move closer to the `peer_manager`, I don't think the concept of a `PeerBook` should mix with the concept of a `Peer`. In other words, I'd leave the `Book`s part of `waku_peer_store` for now.

Reply:
You are totally right, fixed c5a699c6e848cf76d158143605876a5f03c32d55

(I kind of regret using the peerbook, but that's another story :P)